x/registry/types/message_register_top_level_domain.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const TypeMsgRegisterTopLevelDomain = "register_top_level_domain"
+
+var _ sdk.Msg = &MsgRegisterTopLevelDomain{}
+
+func NewMsgRegisterTopLevelDomain(creator string) *MsgRegisterTopLevelDomain {
```

Comment:
Add missing argument `name` and `registrationPeriodInYear`

Reply:
added